doc/tech/approach.md

```diff
@@ -3,97 +3,226 @@
+
+ - Underlay and Overlay Networking concepts
+ - Hardware and Software Infrastructure Layer concepts
+ - Software Defined Networking, SDNu and SDNo concepts
+ - Programmable Networking Fabric concept
+
+<a name="5.1.3.1"></a>
+#### 5.1.3.1 Underlay and Overlay Networking concepts
+
+The ETSI NFV model divides networking into an Underlay and an Overlay Network layer. The purpose with this layering is to ensure separation of the SW Virtualisation tenants (Workload) Overlay Networks from each other, whilst allowing the traffic to flow on the shared Underlay Network in between all Ethernet connected HW units.
+
+The Overlay Networking separation is often done through encapsulation of Tenants' traffic using overlay protocols e.g. through VxLAN or EVPN over the Underlay Network e.g. based on L2 (VLAN) or L3 (IP) networks.
+
+In some instances, the SW Virtualisation Tenants can bypass the Overlay Networking encapsulation to achieve better performance or network visibility/control. A common method to bypass the Overlay Networking encapsulation normally done by the SW Virtualisation Layer, is the VNF/CNF usage of SR-IOV that effectively take over the Physical and Virtual Functions of the NIC directly into the VNF/CNF Tenant. In these cases, the Underlay Networking must handle the separation e.g. through a Virtual Termination End Point (VTEP) that encapsulate the Overlay Network traffic.
+
+> **Note:** Bypassing the Overlay Networking layer is a violation of the basic CNTT decoupling principles, but in some cases unavoidable with existing technologies and standards. Until suitable technologies and standards are developed, CNTT have a set of agreed exemptions that forces the Underlay Networking to handle the bypassed Overlay Networking separation.
+
+VTEP could be manually provisioned in the Underlay Networking or be automated and controlled through a Software Defined Networking controller interfaces into the underlying networking in the HW Infrastructure Layer. 
+
+<a name="5.1.3.2"></a>
+#### 5.1.3.2 Hardware and Software Infrastructure Layer concepts
+
+The Cloud Infrastructure can be considered to be composed of two distinct layers, referred to as HW Infrastructure Layer and SW Infrastructure Layer. When there are multiple separated simultaneously deployed SW Virtualisation domains, the architecture and deployed implementations must enable each of them to be in individual non-dependent administrative domains. The HW Infrastructure must also be enabled to be a fully separated administrative domain from all of the SW Virtualisation domains. 
+
+For Cloud implementations of multiple well separated simultaneous SW Virtualisation domains on a shared HW Infrastructure there must be a separation of the hardware resources e.g. servers and the Underlay Networking resources that interconnect the hardware resources e.g. through a switching fabric.
```

Comment:
"...there must be a separation of the hardware resources..." - Not sure I follow. Which h/w resources?

Reply:
@markshostak When using multiple SW Virtualization domains on a shared pool of physical HW resources in the HW Infrastructure (Compute, Storage and Networking as the offered composed HW resources) there have to be a strict “share nothing” in each HW resource.

One complication of that statement is that a “HW resource” is not always a physical component or box, so I subdivide the explanations a bit below.
- Compute: Today a traditional Server with one or more CPUs, RAM, local disk drives and local PCIe devices is not possible to subdivide into multiple HW separated composed compute server resources that do not share any part of the execution engines. This will though be technically possible in the not so distant future, and we shall ensure that CNTT do not block these possibilities.
- Storage: Today there exist mature technology to assign parts of pooled disk drives on the level of drives, volumes and blocks reachable over “SAS channels”, “NVMe over PCIe” or “NVMe over Ethernet” (often in boxes referred to as JBOD, JBOF or storage servers) into one or more (composed) servers.
This could be used for the purpose of composed servers with flexible amount of HW-level storage (drives, volumes or blocks) or multi-server access to HW-level storage e.g. for redundancy or performance reasons in a storage oriented application.
- Networking: Today the Physical Ethernet Fabric (set of interconnected switches acting as a fabric) is normally shared in between all physical Servers, Storage boxes and external gateways. To achieve a share nothing deployment on these fabrics the HW Infrastructure must ensure that the methods of separation e.g. VLAN selection or VxLAN encapsulation cannot be overruled by mistake or abuse of the connected HW resources that are being used by other administrative domains in the SW Virtualization or Bare Metal application Layer. So on the shared Ethernet, the VLAN VID-ranges or VxLAN VNI-ranges are the hardware resources that cannot be shared in between the SW Virtualization domains.

To exemplify this in a traditional deployment, a Switch Port of a Server NIC must ensure e.g. relevant VLAN filtering or VxLAN encapsulation.

A slightly more complex example is the usage of SmartNICs that are managed from the HW Infrastructure Layer that can ensure e.g. relevant VLAN filtering or VxLAN encapsulation before the packets reach the physical switches. In this example the SmartNIC HW infrastructure ensure the separation of the exemplified VLAN VID-ranges or VxLAN VNI-ranges in a way that is protected from being reachable by the SW Virtualization domains.

Hope this explained some of the facets of a HW resource and why it cannot be an oversimplified example such as “a server box” in all cases.

Reply:
@TFredberg, good explanation.  To make it a bit more clear, I have changed to: "For Cloud implementations of multiple well separated simultaneous SW Virtualisation domains on a shared HW Infrastructure there must be a relevant separation of the hardware resources; for example, the separation of servers and the Underlay Networking resources through a switching fabric." Hopefully it helps;